### PR TITLE
Upgrade `protoc-go-gen-flags` to `v1.0.5`

### LINKED ISSUE #5781: Upgrade `protoc-go-gen-flags` to `v1.0.5`
### Summary

Upgrade `protoc-go-gen-flags` to `v1.0.5`.

### Current Situation

_No response_

### Why do we need this? Who uses it, and when?

This allows for the use of the `semantic` flag on messages which in turn allows for generating CLI flags for messages which have meaning only by existing. For example: [`DynamicMode`](https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.21/api/end_device.proto#L236-L238)

### Proposed Implementation

_No response_

### Contributing

- [X] I can help by doing more research.
- [X] I can help by implementing the feature after the proposal above is approved.
- [X] I can help by testing the feature before it's released.

### Code of Conduct

- [X] I agree to follow TTN's [Community Code of Conduct](https://github.com/TheThingsNetwork/Manifesto/blob/master/CodeOfConduct.md).

### PR DESCRIPTION
**Pull request by ysmilda**
#### Summary

Closes #5781 

#### Changes

- Upgrade `protoc-go-gen-flags` to `v1.0.5`
- Add `semantical` flag to end device `ADRSettings` 
- Regenerate proto flags.

#### Testing

UT.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ For details about compatibility between different releases, see the **Commitment
 - The Network Server now correctly handles the command that may succeed a `LinkADRAns` response.
 - LR-FHSS data rate matching.
 - Console data rate rendering of non-LoRa modulations.
+- `mode.dynamic`, `mode.static`, and `mode.disabled` flags available for `ADRSettings`
```
Comment (adriansmares) [inline CHANGELOG.md]:
```
`--mac-settings.adr.mode.disabled`, `--mac-settings.adr.mode.dynamic` and `--mac-settings.adr.mode.static` flags of the `end-device update` command.
```

**Review by adriansmares — APPROVED** [review]
Tested and works as expected !

File: api/end_device.proto [hunk 2]
```diff
@@ -234,7 +234,7 @@ message ADRSettings {
   // Configuration options for cases in which ADR is to be disabled
   // completely.
   message DisabledMode {
-    option (thethings.flags.message) = { select: true, set: true };
+    option (thethings.flags.message) = { select: true, set: true, semantical: true };
```
Comment (adriansmares) [inline api/end_device.proto]: Is this `semantical` required here given that the parent contains it too ? The generated code stays the same if I remove it.

Comment (ysmilda) [reply api/end_device.proto]: You're correct in that is not necessary, it's an artifact from me playing around a bit. Will remove it.

**Pull request merged**